docs/framework/social-post-generator.mdx

```diff
@@ -2,218 +2,448 @@
+  Drag a **Section** component from the **Toolbox** panel and place it _inside_ the **Section** that's already there, just below the **Message** component.
 
-Next, drag a Tags component from the Content section of the Toolkit into the new Section.  
+  <Note>That's right — **Sections** can contain other **Sections**!</Note>
 
-Then, drag a Separator component below the Tags component. 
+  Select the **Section** you just added. In the **Properties** panel:
 
-Finally, drag a Text component below the button. This is where the text generated by the LLM will appear.
+  - Find the **Title** property and clear it its value to remove the **Section**'s title.
+  - Scroll down to the **Style** section and look for the **Container background** property, which sets the **Section**'s background color. 
+  - Click its **CSS** button, which will cause a text field to appear below it.
+  - Enter this color value into the text field: `#F6EFFD`.
+  </Step>
 
-![Initial UI layout](/framework/images/tutorial/social_post/sp_gen_4.png)
+  <Step title="Add a Tags component">
+  Writer Framework has a number of useful components to make your apps more functional and beautiful. One of these is the **Tags** component, which can take a list of hashtags (or words, or short phrases) and display them inside colorful "bubbles" to make them stand out. This app will display the social media tags it generates in a **Tags** component.
 
+  Drag a **Tags** component from the **Toolbox** panel and place it inside the new **Section**.
+  </Step>
+
+  <Step title="Add a Separator">
+  Drag a **Separator** component from the **Toolbox** panel and place it inside the new **Section**, just below the **Tags** component. This will separate the tags from the posts.
+  </Step>
+
+  <Step title="Add a Text component">
+  Finally, drag a **Text** component from the **Toolbox** panel and position it below the **Separator**. This will hold the generated social media posts.
+  </Step>
+</Steps>
+
+The project editor should now look like this:
+
+![Project editor with components laid out](/framework/images/tutorial/social_post/sp_gen_2c.png)
 
-## Updating the code
 
-With the UI set up, you’ll next add some code to manage user input, communicate with an LLM when the user interacts with the UI, and update the application state based on the results.
+## Add the back-end code
+
+With the UI laid out, it's time to work on the logic behind it. 
+
+The logic behind the user interface is defined in a file named `main.py`, which is in your project's directory. This file was automatically generated; you'll update the code in it to define the behavior of your app.
+
+There are two ways to edit `main.py`:
+
+1. **In your browser.** Click on the **Code** button near the top right corner of the project editor page. A pane with the name **Code**  will appear at the bottom half of the screen, displaying and allowing you to edit the contents of `main.py`.
```

Comment:
>near the top right corner

When updating screenshots, please make sure that these references are also updated.